parser.go

```diff
@@ -15,8 +16,15 @@ type parser struct {
 }
 
 func newParser(s string) *parser {
+	var fixedUp string
+	{
+		// remove any leading/trailing whitespace
+		fixedUp = strings.TrimSpace(s)
+		// remove any leading space before a right parenthesis (issue #42)
+		fixedUp = strings.ReplaceAll(fixedUp, " )", ")")
```

Comment:
This doesn't handle the case where there's more than 1 whitespace before the next parens. I think we may need a loop here to consume the whitespaces.

Reply:
ty!